action/protocols/vote/protocol.go

```diff
@@ -32,46 +41,95 @@ func (p *Protocol) Handle(act action.Action, ws state.WorkingSet) error {
 	if !ok {
 		return nil
 	}
-	voteFrom, err := ws.LoadOrCreateAccountState(vote.Voter(), big.NewInt(0))
+	// Get candidateList from trie and convert it to candidates map
```

Comment:
here should be same as workingset.go code now:

if ws.Height() > 0 {
    candidates, err := ws.getCandidates(blockHeight - 1)
    ....
}